### PR TITLE
Update to JupyterLab 3.0 RC

### LINKED ISSUE #832: JupyterLab 3.0 Release (umbrella issue)
JupyterLab's 3.0 release will be based on Jupyter Server - rather than Notebook - among other things.  This issue is created to track issues related to that effort.

Back-end tasks:
- [x] Convert server tests to Jupyter Server's framework #831
- [x] Rework server extension interface to Jupyter Server #834

Front-end tasks:
- [x] Clean up build
    - [x] Remove redundant build steps
    - [x] Figure out if new build files can be created dynamically on build (look into how core does it)
- [x] Get theme extension working
- [x] Get pipeline extension working
- [x] Test functionality that appears to be working 
- [x] Lab icon redundant import issue has returned

Issues found during integration:
- [x] Run and Export buttons are disabled on pipelines #1010
- [x] Notebook and Python script files cannot be dragged onto canvas #1012
- [x] Notebook and Python files cannot be opened by double-clicking node #1015

### PR DESCRIPTION
**Pull request by ajbozarth**
Update Elyra to run on JupyterLab 3.0 RC

The following temporary changes will need to be reverted prior to the next release:
- adding `--pre` to applicable pip installs
- commenting out dependancies and installations of the git extension

Fixes #832

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by ajbozarth** [comment]
As I was opening this I noticed flake8 was commented out in the Makefile. I'm currently checking if that needs to be addressed and will push an update if it works as is, otherwise fixing flake will be a followup issue

**Comment by jtpio** [comment]
Just curious: do you also plan to adopt the new distribution system in lab 3.0 to distribute each extension separately as Python packages?

Or bundle them all in the main `elyra` package similar to the current approach?

https://github.com/elyra-ai/elyra/blob/19a484ffdff6a717ae7bb1eb6c5db2d017d66e6a/setup.py#L102

**Comment by ajbozarth** [comment]
> do you also plan to adopt the new distribution system in lab 3.0 

We plan to initially continue using npm for our extension packages, as seen in this PR, to release 3.0 support asap. Then in a future release we plan to update to the new federated extensions, we have not decided how soon that update will happen though.

File: setup.py [hunk 1]
```diff
@@ -54,23 +54,26 @@
         'jsonschema>=3.2.0',
         'jupyter_core>=4.0,<5.0',
         'jupyter_client>=6.1',
-        'jupyterlab-git==0.23.1',
-        'jupyterlab>=2.0.0,<3.0.0',
+        'jupyter_server>=1.0.5',
+        'jupyterlab>==3.0.0rc9',
```
Comment (karlaspuldaro) [inline setup.py:58]: should we be using rc10?
```suggestion
        'jupyterlab>==3.0.0rc10',
```

Comment (ajbozarth) [reply setup.py:58]: if you've tested that things still work on rc10 feel free to commit this, I'm in the middle of some coding and can't check out and test it rn

Comment (karlaspuldaro) [reply setup.py:58]: I haven't tested out but I can help looking into that

Comment (lresende) [reply setup.py:58]: I have been using rc10, will commit it

**Review by lresende — APPROVED** [review]

**Pull request merged**